imgui_plot.h

```diff
@@ -1102,14 +1105,32 @@ namespace ImGui
             }
             float const * const y = y_data_bottoms ? y_data_bottoms : y_data;
 
+            std::vector<ImVec2> v;
+
             ImDrawList* dl = window->DrawList;
-            dl->PathLineTo(ImVec2(figure.GetScalesX()->Scale(x_data[0]), figure.GetScalesY()->Scale(y[0] - y_data[0])));
+            v.emplace_back(figure.GetScalesX()->Scale(x_data[0]), figure.GetScalesY()->Scale(y[0] - y_data[0]));
+            dl->PathLineTo(v.back());
 
             for (size_t i = 0; i < n_data; ++i) {
-                dl->PathLineTo(ImVec2(figure.GetScalesX()->Scale(x_data[i]), figure.GetScalesY()->Scale(y[i])));
+                v.emplace_back(figure.GetScalesX()->Scale(x_data[i]), figure.GetScalesY()->Scale(y[i]));
+                dl->PathLineTo(v.back());
+            }
+
+            v.emplace_back(figure.GetScalesX()->Scale(x_data[n_data-1]), figure.GetScalesY()->Scale(y[n_data-1] - y_data[n_data-1]));
+            dl->PathLineTo(v.back());
+
+            ImVec2 pointer(GImGui->IO.MousePos.x, GImGui->IO.MousePos.y);
+
+            const bool check = is_inside_area(pointer, v);
+
+            if (check && properties_.tooltip) {
+                SetTooltip("%s", properties_.tooltip);
+            }
+
+            if (check && properties_.area_hover_col) {
+                dl->PathFillConvex(properties_.area_hover_col);
             }
 
-            dl->PathLineTo(ImVec2(figure.GetScalesX()->Scale(x_data[n_data-1]), figure.GetScalesY()->Scale(y[n_data-1] - y_data[n_data-1])));
             dl->PathFillConvex(properties_.area_fill_col);
```

Comment:
Have you tried this code with data that has both positive and negative values?  Unless I am missing it, I don't see a `PathLineTo` call that goes from the X-axis to the first data point, the last data point to the X-axis, and then a final line that goes from the X-axis at the first data point to the X-axis at the last data point (i.e., to complete the polygon).

Reply:
There are two calls to `PathLineTo()` for those points, one above the `for` loop, and one below it.

Maybe they were not obvious because, instead of showing something like `0` for the y value, I do:
`y[0] - y_data[0]`
which, in case `y` points to `y_data`, it simply means `0`. But, in case `y` points to `y_data_bottoms`, then the value will be `0 + the previous stack y value`.

Does this answer your question?

Reply:
Perfect.  I missed that.